doc/developer/platform/ux.md

```diff
@@ -772,6 +772,36 @@ running will be hidden from users. Any log messages in `materialized` today that
 are meant for end-users will need to instead make their messages available via
 tables or sources in the system catalog.
 
+### Correctness
+
+Materialize maintains three correctness guarantees.
+
+1. *Transactions in Materialize are strictly serializable with 
+   respect to the operations that occur inside of Materialize.* 
+   Operations include:
+    * `SELECT`, `INSERT`, `UPDATE`, and `DELETE` statements (but not `TAIL`)
+    * Materialize initiated acknowledgements of upstream sources (*e.g., the 
+      commit of an offset by a Kafka source, the acknowledge of an LSN by a 
+      PostgresSQL source, etceterea*)
+2. *Materialize respects the explicit or implied event order of its sources.  
+   This includes partial orders*  In practice this means Materialize assigns 
+   new timestamps to sources.  This assignment of new timestamps is called 
+   reclocking.  For sources without transactional semantics (basic Kafka streams)
+   Maerialize reclocks the events to reflect the order in which the source emits 
+   each event. For sources that do have transactional semantics (Postgres, Debezium) 
+   Materialize reclocks events to respect the transactional boundaries.  For partially 
+   ordered elements x and y:
+    1.  if *x < y* in the source, then *x ≤ y* in Materialize
+    2.  if *x = y* in the source, then *x = y* in Materialize
```

Comment:
One additional thing we may want to mention is that this is materialize commits to exactly one order of events that respects that source's implied/explicit/partial order; I.e.

this is a consequence of the combination of #1 and #2, so its your call on whether we should mention this property that falls out of those, so the reader doesn't need to figure that out themselves!

Reply:
I added a note about reclocking being a permanent order, which implies the exactly once idea you're mentioning.